lib/tools/adb-commands.js

```diff
@@ -110,8 +110,16 @@ methods.keyevent = async function (keycode) {
 };
 
 methods.lock = async function () {
-  log.debug("Pressing the KEYCODE_POWER button to lock screen");
-  await this.keyevent(26);
+  let locked = await this.isScreenLocked();
+  if (!locked) {
+    log.debug("Pressing the KEYCODE_POWER button to lock screen");
+    await this.keyevent(26);
+    // we need to wait for the animation to complete before
+    // the screen is actually considered 'locked'
+    await sleep(500);
```

Comment:
maybe add a TODO here in the vein of "TODO: we should probably spin on `isScreenLocked` instead of a static wait"

.... or just do it :-)